src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/SignatureHelper.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+
+namespace System.Reflection.Emit
+{
+    // TODO: Only support simple signatures. More complex signatures will be added.
+    internal static class MetadataSignatureHelper
+    {
+        internal static BlobBuilder FieldSignatureEncoder(Type fieldType)
+        {
+            var fieldSignature = new BlobBuilder();
+
+            WriteSignatureTypeForReflectionType(new BlobEncoder(fieldSignature).FieldSignature(), fieldType);
+
+            return fieldSignature;
+        }
+
+        internal static BlobBuilder MethodSignatureEncoder(ModuleBuilderImpl _module, Type[]? parameters, Type? returnType, bool isInstance)
+        {
+            // Encoding return type and parameters.
+            var methodSignature = new BlobBuilder();
+
+            ParametersEncoder parEncoder;
+            ReturnTypeEncoder retEncoder;
+
+            new BlobEncoder(methodSignature).
+                MethodSignature(isInstanceMethod: isInstance).
+                Parameters((parameters == null) ? 0 : parameters.Length, out retEncoder, out parEncoder);
+
+            if (returnType != null && returnType != _module.GetTypeFromCoreAssembly("System.Void"))
+            {
+                WriteSignatureTypeForReflectionType(retEncoder.Type(), returnType);
+            }
+            else // If null mark ReturnTypeEncoder as void
+            {
+                retEncoder.Void();
+            }
+
+            if (parameters != null) // If parameters null, just keep the ParametersEncoder empty
+            {
+                foreach (Type parameter in parameters)
+                {
+                    WriteSignatureTypeForReflectionType(parEncoder.AddParameter().Type(), parameter);
+                }
+            }
+
+            return methodSignature;
+        }
+
+        private static void WriteSignatureTypeForReflectionType(SignatureTypeEncoder signature, Type type)
+        {
+            // We need to translate from Reflection.Type to SignatureTypeEncoder. Most common types for proof of concept. More types will be added.
+            switch (type.FullName)
```

Comment:
This switch should be done by comparing Type objects, without fetching FullName.

(This can wait for follow up PR.)